oop1-construct.js

```diff
@@ -1,16 +1,62 @@
-//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.  This string should clearly explain why classes are used in Object Oriented Programming.  The first method is called explain().  This method should print the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that are essential to building a class.  Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
+//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.  This string should clearly explain why classes are used in Object Oriented Programming. 
+// The first method is called explain().  This method should print the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that are essential to building a class.  
+//Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
 
 //your code here...
+class WhyClass {
+    constructor(purpose) {
+
+        this.purpose = purpose;
+    }
+    explain(purpose) {
+        console.log(purpose);
+    }
+    pieces() {
+        console.log('The important pieces of a class are the contructor, getters, setters, methods and attributes.')
```

Comment:
This is fine for this assignment, and achieves the goal. But try and think about writing methods as making a machine to do things for you. And if you ever need it to do a slightly different thing, you should use a variable. So I'd refactor this method so it could console.log a variable that it's passed. `pieces(message) { console.log(message) `}